src/app/utils/xml-utils.ts

```diff
@@ -20,3 +20,12 @@ export function parseXml(xmlStr: string): XMLElement {
   }
   throw new Error('No XML parser found');
 }
+
+export function parseAttributes(xml: XMLElement): AttributesData {
+  const attributes: AttributesData = {};
+  Array.from(xml.attributes).forEach((attr) => {
```

Comment:
Replace `(attr)` with `({name, value})` and use them directly

Reply:
You can also remap attributes with `xml` namespace, such as `xml:id` or `xml:lang`, in order to remove the `xml:` prefix. 
Moreover, use this function in `GenericParserService` instead of its private `getAttributes`: they have the same purpose.